src/Rector/AbstractRector.php

```diff
@@ -256,6 +263,11 @@ final public function enterNode(Node $node)
             $rectorWithLineChange = new RectorWithLineChange($this::class, $originalNode->getLine());
             $this->file->addRectorClassWithLine($rectorWithLineChange);
 
+            // refresh PHPStan scope, parent connections and other attributes
+            $scope = $node->getAttribute(AttributeKey::SCOPE);
+            $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
+            $this->phpStanNodeScopeResolver->refreshStmtNodes($parent, [$node], $scope);
+
             // update parents relations - must run before connectParentNodes()
```

Comment:
This is the important change. On changed node, all the children of this node and this node get retraverse of scope.